test/test_distributed.py

```diff
@@ -168,6 +168,27 @@ def _init_global_test(self):
         rank = dist.get_rank()
         return (group, group_id, rank)
 
+    # HELPER FOR MULTIGPU TESTS
+    def _init_multigpu_helper(self):
+        """Multigpu tests are designed to simulate the multi nodes with multi
+        GPUs on each node. Nccl backend requires equal #GPUs in each process.
+        On a single node, all visible GPUs are evenly
+        divided to subsets, each process only uses a subset.
+        """
+        nGPUs = torch.cuda.device_count()
+        world_size = dist.get_world_size()
+        visible_devices = range(nGPUs)
+
+        if BACKEND == 'nccl':
+            apply_hack_for_nccl()
+
+        nGPUs_per_process = int(nGPUs / world_size)
+        rankToGPUMapping = {}
+        for i in range(world_size):
+            rankToGPUMapping[i] = visible_devices[
+                i * nGPUs_per_process: (i + 1) * nGPUs_per_process]
```

Comment:
This will be a `range` object in Python3, and it may give some unexpected results. Please convert it to a list explicitly